core/templates/dev/head/domain/learner_dashboard/LearnerPlaylistService.js

```diff
@@ -0,0 +1,112 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service related to the learner playlist.
+ */
+
+ oppia.factory('LearnerPlaylistService', [
+  '$http', '$modal', 'alertsService', 'UrlInterpolationService',
+  function($http, $modal, alertsService, UrlInterpolationService) {
+    var _addToLearnerPlaylist = function(activityId, activityType) {
+      var successfullyAdded = true;
+      var addToLearnerPlaylistUrl = (
+        UrlInterpolationService.interpolateUrl(
+          '/learnerplaylistactivityhandler/<activityType>/<activityId>', {
+            activityType: activityType,
+            activityId: activityId
+          }));
+      $http.post(addToLearnerPlaylistUrl, {})
+        .then(function(response) {
+          if (response.data.belongs_to_completed_or_incomplete_list) {
+            successfullyAdded = false;
+            alertsService.addInfoMessage(
+              'You have already completed or are completing this ' +
+              'activity.');
+          }
+          if (response.data.belongs_to_subscribed_activities) {
+            successfullyAdded = false;
+            alertsService.addInfoMessage(
+              'This is present in your creator dashboard');
+          }
+          if (response.data.playlist_limit_exceeded) {
+            successfullyAdded = false;
+            alertsService.addInfoMessage(
+              'Your \'Play Later\' list is full!  Either you can ' +
+              'complete some or you can head to the learner dashboard ' +
+              'and remove some.');
+          }
+          if (successfullyAdded) {
+            alertsService.addSuccessMessage(
+              'Successfully added to your \'Play Later\' list.');
+          }
+        });
+      return successfullyAdded;
+    };
+
+    var _removeFromLearnerPlaylist = function(activityId, activityTitle,
```

Comment:
Here and elsewhere, when stuff in (...) spans more than one line, break after first paren.

Reply:
Done!